extensions/notification/NotificationQueuePopulator.js

```diff
@@ -38,6 +39,45 @@ class NotificationQueuePopulator extends QueuePopulatorExtension {
             || key === undefined);
     }
 
+    /**
+     * Removes versionId from key
+     * @param {string} key object key
+     * @returns {string} object key without versionId
+     */
+    _getObjectKeyNoVersion(key) {
+        return key.split(VID_SEP)[0];
+    }
+
+    /**
+     * Decides if the entry should be processed based
+     * on its versioning configuration
+     *
+     * Bucket not versioned:
+     * Process only master key events for that bucket
+     *
+     * Bucket has versioning active:
+     * Only process version events for that bucket
+     *
+     * Bucket has versioning suspended
+     * Process both version and master events for that bucket
+     *
+     * @param {*} key object key
+     * @param {*} versioningConfig bucket versioning configuration
+     * @returns {Boolean} should the entry be processed
+     */
+    _shouldProcessEntry(key, versioningConfig) {
+        // bucket versioning suspended condition (accept all)
+        const versionSuspendedCond = versioningConfig &&
+            versioningConfig.Status === 'Suspended';
+        // bucket versioning active condition (accept versions)
+        const versionActiveCond = versioningConfig &&
+            versioningConfig.Status === 'Enabled' && !isMasterKey(key);
```

Comment:
Beware, as versioned buckets can contain objects with only a master key and no version key, for objects written before versioning was enabled. For notification purpose it may not matter much but we may lose notifications around the time a bucket's versioning is enabled due to this. Can you check this to make sure this is wanted?

Maybe we would need to process all keys and ignore versions when we see a master instead.

Reply:
is there an easy way to identify the "last" version of an object?
just ignoring changes to this last version (and processing master instead) would deal with all situations I guess...

Reply:
Not that i know of, i checked the metadata we store for versions and none contain that information

Reply:
the "last" version is the first version key listed for a particular object key. The latest version ID is also stored in the master key if it helps.